posthog/queries/util.py

```diff
@@ -171,3 +171,25 @@ def start_of_week_fix(filter: Filter) -> str:
     This function adds mode to the trunc_func, but only if the interval is week
     """
     return "0," if filter.interval == "week" else ""
+
+
+# TODO: Improve the typing here
+def filter_with_search_properties(filter: Any) -> Any:
```

Comment:
I'd suggest putting this in `actor_base_query.py`.

The `filter` type ought to be `Union[ all existing filters]`.

(see: https://github.com/PostHog/posthog/blob/master/posthog/queries/column_optimizer/foss_column_optimizer.py#L25 )

which implies adding `searchMixin` to all filters, which makes sense, since everything ought to _just work_, since we're making the change to `ActorBaseQuery` which is used in all insights. (and subsequently, would be worth confirming that the modal search works in all insights)